src/shogun/labels/MappedBinaryLabels.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Heiko Strathmann
+ */
+
+#include <shogun/labels/MappedBinaryLabels.h>
+#include <shogun/labels/MulticlassLabels.h>
+
+
+using namespace shogun;
+
+std::tuple<LabelMap, LabelMap> CMappedBinaryLabels::create_mapping(const CLabels* orig) const
+{
+	LabelMap to_internal;
+	LabelMap from_internal;
+
+	switch (orig->get_label_type())
+	{
+		case LT_BINARY:
+			break;
+		case LT_MULTICLASS:
+		{
+			auto mc = (const CMulticlassLabels*) orig; // TODO make const version of .as<>
```

Comment:
what happens when you use .as<> with const? Does it remove qualifiers?

Reply:
doesnt work, would need to be added. but also I guess the static cast does the job here in the switch?